knimin/lib/data_access.py

```diff
@@ -4218,6 +4222,24 @@ def read_shotgun_plate(self, plate_id):
             res = dict(res[0])
             res['condensed_plates'] = [eval(v)
                                        for v in eval(res['condensed_plates'])]
+            # Get positions
+            sql = """SELECT row, col, sample_id, dna_concentration
+                     FROM pm.shotgun_plate_layout
+                     WHERE shotgun_plate_id = %s"""
+            TRN.add(sql, [plate_id])
+            values = TRN.execute_fetchindex()
+            # getting rows/cols and discarding them from res
+            rows = res['rows']
+            del res['rows']
+            cols = res['cols']
+            del res['cols']
+            res['shotgun_plate_layout'] = [[None for c in range(cols)]
```

Comment:
Saw that but here I preferred to have all values in a single "matrix" as dicts vs multiple matrices